api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -127,6 +126,22 @@ public Workspace get(String ns, String firecloudName) {
                 fcWorkspace -> fcWorkspace));
   }
 
+  @Override
+  public Map<User, WorkspaceAccessEntry> getFirecloudWorkspaceAcls(Workspace workspace) {
+    WorkspaceACL firecloudWorkspaceAcls =
+        fireCloudService.getWorkspaceAcl(
+            workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
+    Map<String, Object> aclsMap = (Map) firecloudWorkspaceAcls.getAcl();
+    Map<User, WorkspaceAccessEntry> userToAcl = new HashMap<>();
+    for (Map.Entry<String, Object> entry : aclsMap.entrySet()) {
+      WorkspaceAccessEntry acl =
+          new Gson().fromJson(entry.getValue().toString(), WorkspaceAccessEntry.class);
+      User currentUser = userDao.findUserByEmail(entry.getKey());
```

Comment:
Is it necessary to fetch users like this, or could we just return strings? It's going to get slow for large ACL lists to be hitting the database in a for loop like this.

Reply:
done; this fixed my testing bug 👍